### DISCUSSION TITLE
When Are Component Updates Backported to Stable?

### DISCUSSION THREAD

**Discussion by laurent-indermuehle** [ansible-collections/ansible.mysql#586] [Ideas]
It's not clear to me if stable branches need to be updated when we add a new version of a components (Python, Ansible, MySQL, MariaDB, connector) to the tests. Maybe we could clarify that in the "Releases Support Timeline".

This issue comes up when I wanted to backport something today that devel targets Ansible 2.17 and fails to validate stable-2 because of old syntax that are now deprecated.

I think the confusion, for me, comes from the fact that I don't know for whom we maintain stable branches. Is it for people:
- that don't have time to update their playbooks and can't update their ansible stack
- that don't have time to update their playbooks but can update their ansible stack
- that have time to update their playbooks but can't update their ansible stack

Stable-2 is EoL in 1 month, but the question remains for the upcoming stable-3.

**Comment by Andersson007** [comment]
We support stable-* branches (including testing against core, etc.) as long as they are declared supported in [Releases Support Timeline](https://github.com/ansible-collections/community.mysql#releases-support-timeline).
When a change cannot be backported, e.g. it relates to a new module / new feature / other things not supported by a stable-* branch, we don't backport them.

Though I'm not sure I fully understand the issue.
 
If you'd like to add any clarifications, please submit a PR and let's discuss the wording.

**Comment by laurent-indermuehle** [comment]
Let's try to illustrate. We support stable-* branches for 2 years after their next major release. I want to know what happen when new components are released during that time:
```
----- 2 y ----------------- 1 y ----------------- Today
  \---------------stable-2---------------------------- EoL
              ^                 ^              ^
              |                 |              |
            2.15              2.16           devel
           2.12 eol        2.13 eol
```
Does that means because stable-2 was still supported when Ansible 2.15 and 2.16 were release, we need to support them as well?

Same applies for Python, MySQL, MariaDB and connectors. But I think the same decision will apply to them, correct me if I'm wrong.

Also what about deprecation of older Ansible version. Before `main` becomes `stable-2` it was supporting 2.12 and 2.13. The deprecation occurred while the branch is supported. Should we keep or stop testing?

To summarize, what `stable-2` should support:

- A: 2.12, 2.13, 2.14
- B: 2.15, 2.16, devel
- C: 2.12, 2.13, 2.14, 2.15, 2.16
- D: 2.12, 2.13, 2.14, 2.15, 2.16, devel
- E: other

This time, it's not only a question of tests. In order to support 2.17, we must edit all modules and options that are now deprecated.

**Reply by Andersson007** [reply]
@laurent-indermuehle ah, cool, I see now, thanks for the detailed explanation!
Great questions! I don't know:)
- What do you think would be the best option (from user's and maintainer's perspective)?
- Do you think if we should discuss this on the [forum](https://forum.ansible.com/) under the `Project Discussions` category (maybe some tags would also suit)?

**Reply by laurent-indermuehle** [reply]
@Andersson007

I think we should lock the supported version at the stable branch creation. Meaning the tests won't change during the 2 years of support.

That means if a fix uses a newer syntax, we must revert to a older syntax when backporting to stable.

This clarification will allow us to use all the new features (Like f strings introduced in Python 3.6 for instance...). And it will only hurt use during backports.

But I don't have that much experience on supporting older collections version. I would love to have more input on this. If the forum could provide that, it would be nice.

I remember a forum was created for Postgresql, do we need to ask to do the same for c.mysql?

**Reply by Andersson007** [reply]
@laurent-indermuehle yep, the PostgreSQL forum group exists.
- You can request it in a [special topic](https://forum.ansible.com/t/requesting-a-forum-group/503/15). And request the `mysql` tag or something. You'll be a group owner, so you can fill up the group info similarly to [PostgreSQL one](https://forum.ansible.com/g/PostgreSQLTeam). Let me know after that:
  - I'll add a bot Matrix notifications when related posts are created.
  - we should update our README with the group info
  - invite people in Bullhorn, Matrix and here (current maitainers) and people listed in https://github.com/ansible-collections/community.mysql/wiki/MySQL-Working-Group to join the group
  - we should replace our wiki [WG page](https://github.com/ansible-collections/community.mysql/wiki/MySQL-Working-Group) with a link to the forum group
  - we should update https://docs.ansible.com/ansible/devel/community/communication.html

- After this is done ^, you could post this on the forum. (though i think it's not an obstacle to post this discussion on the forum w/o the group)

**Reply by Andersson007** [reply]
@laurent-indermuehle i've just finished with the forum group ^.
would you like to start the discussion there and invite people to join using bullhorn?

**Reply by laurent-indermuehle** [reply]
@Andersson007 It's a mix of a community question and a MySQL collection issue. I'll need to rewrite my question. I'll try to find the time to post something on the forum this week. Thanks for the reminder.
